### ISSUE TITLE
[BUG] OSACA cannot parse some instructions generated by AdaptiveCpp/Clang

### ISSUE THREAD

**Issue by MarkusBuettner** [RRZE-HPC/OSACA#107]
**Describe the bug**
For this test, I compiled a simple vector triad written in SYCL with AdaptiveCpp (with Clang 15):
`/opt/AdaptiveCpp/bin/acpp -march=icelake-server -O3 -ffast-math --acpp-targets="omp" -S triad.cpp`
Then, when I feed the output file to osaca I get an error 
`ValueError: Could not parse instruction on line 28974: '\tcmpb\t$0, %fs:_ZGVZN7hipsycl4glue6detail13random_numberImEET_vE2rd@TPOFF'`

Commenting this instruction results in an error further down. The common factor appears to be the ` %fs:_ZGVZN7hipsycl4glue6detail13random_numberImEET` part.

**To Reproduce**
|||
|---|---|
| OSACA version | v 0.5.3 |
| Used where | CLI, installed through pip

- OSACA command: `osaca --ignore-unknown --arch ICX --lines 38678-38697 triad.s`
- relevant section from assembly file ([full file attached](https://github.com/user-attachments/files/16335300/triad.txt)): 
```asm
28971         movq    %rdi, %r15
28972         callq   _ZNSt6chrono3_V212system_clock3nowEv@PLT
28973         movq    %rax, %r14
28974         cmpb    $0, %fs:_ZGVZN7hipsycl4glue6detail13random_numberImEET_vE2rd@TPOFF
28975         je      .LBB158_1
28976 # %bb.5:
28977         cmpb    $0, %fs:_ZGVZN7hipsycl4glue6detail13random_numberImEET_vE3gen@TPOFF
28978         je      .LBB158_6
28979 .LBB158_13:
28980         cmpb    $0, %fs:_ZGVZN7hipsycl4glue6detail13random_numberImEET_vE12distribution@TPOFF
28981         je      .LBB158_14
28982 .LBB158_15:
28983         movq    %r14, %rbx
28984         shrq    $56, %rbx
28985         movq    %fs:0, %rax

```

**OSACA output**
```
Traceback (most recent call last):
  File "/home/buettner/.local/lib/python3.10/site-packages/osaca/parser/parser_x86att.py", line 259, in parse_line
    result = self.parse_instruction(line)
  File "/home/buettner/.local/lib/python3.10/site-packages/osaca/parser/parser_x86att.py", line 277, in parse_instruction
    result = self.instruction_parser.parseString(instruction, parseAll=True).asDict()
  File "/usr/lib/python3/dist-packages/pyparsing.py", line 1955, in parseString
    raise exc
  File "/usr/lib/python3/dist-packages/pyparsing.py", line 3814, in parseImpl
    raise ParseException(instring, loc, self.errmsg, self)
pyparsing.ParseException: Expected end of text, found ':'  (at char 23), (line:1, col:24)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/buettner/.local/bin/osaca", line 8, in <module>
    sys.exit(main())
  File "/home/buettner/.local/lib/python3.10/site-packages/osaca/osaca.py", line 462, in main
    run(args, output_file=args.out)
  File "/home/buettner/.local/lib/python3.10/site-packages/osaca/osaca.py", line 411, in run
    inspect(args, output_file=output_file)
  File "/home/buettner/.local/lib/python3.10/site-packages/osaca/osaca.py", line 332, in inspect
    raise e
  File "/home/buettner/.local/lib/python3.10/site-packages/osaca/osaca.py", line 318, in inspect
    parsed_code = parser.parse_file(code)
  File "/home/buettner/.local/lib/python3.10/site-packages/osaca/parser/base_parser.py", line 58, in parse_file
    asm_instructions.append(self.parse_line(line, i + 1 + start_line))
  File "/home/buettner/.local/lib/python3.10/site-packages/osaca/parser/parser_x86att.py", line 261, in parse_line
    raise ValueError(
ValueError: Could not parse instruction on line 28974: '\tcmpb\t$0, %fs:_ZGVZN7hipsycl4glue6detail13random_numberImEET_vE2rd@TPOFF'
```

**Expected behavior**
I would expect that this instruction can be parsed properly, the `%fs:` prefix does not cause issues if followed by a number (e.g. in `movq    %fs:0, %rax`)

**Comment by MarkusBuettner** [comment]
So I did some more digging, the problem is apparently in the parsing of the segment extensions.
Parsing `%fs:_ZGVZN7hipsycl4glue6detail13random_numberImEET_vE2rd@TPOFF` fails, but `%fs:_ZGVZN7hipsycl4glue6detail13random_numberImEET_vE2rd@TPOFF()` is fine. 
I guess having the base, index and scale parameters inside the brackets is optional, but then the brackets should also be optional. 

I changed the segment_extension part in ParserX86ATT (line 97ff) to make the brackets optional, and that statement is now parsed correctly.

```python
       segment_extension = (
           hex_number
            ^ pp.Word(pp.nums)
            ^ pp.Group(
                 pp.Optional(offset.setResultsName("offset"))
                + pp.Optional(pp.Literal("(")
                + pp.Optional(self.register.setResultsName("base"))
                + pp.Optional(pp.Suppress(pp.Literal(",")))
                + pp.Optional(self.register.setResultsName("index"))
                + pp.Optional(pp.Suppress(pp.Literal(",")))
                + pp.Optional(scale.setResultsName("scale"))
                + pp.Literal(")"))
            )
        )
```

Another case is a few lines down (~29094)  for identifiers
```
_ZZN7hipsycl4glue6detail13random_numberImEET_vE3gen@TPOFF-8
_ZZN7hipsycl4glue6detail13random_numberImEET_vE3gen@TPOFF+4992
```
although the first one seems to only fail in a separate test suite.

For this, I changed the parsing of the identifier to also support offsets at the end:
```python
         self.identifier = identifier = pp.Group(
             pp.Optional(id_offset).setResultsName("offset")
             + pp.Combine(
                 pp.delimitedList(pp.Combine(first + pp.Optional(rest)), delim="::"),
                 joinString="::",
             ).setResultsName("name")
             + pp.Optional(relocation).setResultsName("relocation")
             + pp.Optional(pp.Suppress(pp.Optional(pp.Literal("+"))) + decimal_number).setResultsName("offset")
         ).setResultsName("identifier")
```

and now OSACA is able to parse the provided sample file successfully.

I will prepare a pull request in the coming days for review.

**Comment by JanLJL** [comment]
Fixed with PR #108
